File: app/src/main/java/com/wavesplatform/wallet/ui/dex/details/chart/CustomCandleChart.java
package com.wavesplatform.wallet.ui.dex.details.chart;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

import com.github.mikephil.charting.charts.CandleStickChart;

public class CustomCandleChart extends CandleStickChart {

    private ViewPager view;

    public CustomCandleChart(Context context) {
        super(context);
    }

    public CustomCandleChart(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomCandleChart(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        int action = ev.getAction();
        switch (action) {
            case MotionEvent.ACTION_DOWN:
                view.requestDisallowInterceptTouchEvent(true);
                break;

            case MotionEvent.ACTION_UP:
                view.requestDisallowInterceptTouchEvent(false);
                break;
        }

        super.onTouchEvent(ev);
        return true;
    }

    public void setView(ViewPager view) {
        this.view = view;
    }
}

File: app/src/main/java/com/wavesplatform/wallet/v2/ui/custom/CustomCandleChart.kt
/*
 * Created by Eduard Zaydel on 1/4/2019
 * Copyright © 2019 Waves Platform. All rights reserved.
 */

package com.wavesplatform.wallet.v2.ui.custom

import android.content.Context
import android.support.v4.view.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent

import com.github.mikephil.charting.charts.CandleStickChart

class CustomCandleChart : CandleStickChart {

    private var view: ViewPager? = null

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {}

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        val action = ev.action
        when (action) {
            MotionEvent.ACTION_DOWN -> view?.requestDisallowInterceptTouchEvent(true)

            MotionEvent.ACTION_UP -> view?.requestDisallowInterceptTouchEvent(false)
        }

        super.onTouchEvent(ev)
        return true
    }

    fun setView(view: ViewPager) {
        this.view = view
    }

    fun onDestroy() {
        this.view = null
    }
}
